File: src/game/tools/rpc/output/wrap/QuestDataWrap.go
package wraps

import (
	rpc "tipcat.com/common/rpc/simple"
)

type QuestDataWrap struct {
	
	QuestID int32 // 关卡ID
	ChestState bool // 宝箱是否发放
	LastUpdateTime int64 // 最后一次更新时间
	rpc.Wrapper
}

func (w *QuestDataWrap)Decode(pck *rpc.Packet) rpc.Wrapper {
	
	w.QuestID = pck.PopInt32(); 
	w.ChestState = pck.PopBool(); 
	w.LastUpdateTime = pck.PopInt64(); 
	return w
}

func (w *QuestDataWrap)Encode(pck *rpc.Packet) {
	
	pck.PutInt32(w.QuestID) 
	pck.PutBool(w.ChestState) 
	pck.PutInt64(w.LastUpdateTime) 
}

File: src/game/tools/rpc/output/wrap/QuestDataWrap.cs
using System;
using Common.Net.Simple;

namespace Game 
{
	public class QuestDataWrap : IWrapper
	{	
		public int QuestID; // 关卡ID
		public bool ChestState; // 宝箱是否发放
		public long LastUpdateTime; // 最后一次更新时间
		
		public void Decode(Packet pck)
		{	
			QuestID = pck.GetInt(); 
			ChestState = pck.GetBool(); 
			LastUpdateTime = pck.GetLong(); 
		}
		
        public void Encode(Packet pck)
        {	
        	pck.PutInt(QuestID); 
        	pck.PutBool(ChestState); 
        	pck.PutLong(LastUpdateTime); 
        }
	}
}